### PR TITLE
User frontend final raw form

### PR DESCRIPTION
**Pull request by dorobat-diana**
Final Raw form of User Interface. User Page containt : most recently played songs+ recommended songs+ advertised songs + backhome button + searchbutton). When clicking on a song a new page is opened (UserSongDashboard) with all the SongBasicInfo. When the search button is clicked a new Search Page is opened. Search Page contains: back to user button (gets you back on the User Page), functional searchbar(it work with a mockdata but I ve put a trigger for text change) and a list with the returned songs (again when clicking on a song you can open its dashboard)

### CODE REVIEW

**Review by DanielCujba — APPROVED** [review]
LGTM👍

File: TechTitans/Views/Components/User/BackToUser.xaml
Comment (mihneadumi) [inline TechTitans/Views/Components/User/BackToUser.xaml:1]: nu e nevoie de back to user daca folosesti navigation ul pe care l am introdus eu in master
ex: Navigation.PushAsync(new ArtistSongDashboard(songInfo));

Atunci iti pune automat sus buton de back si poti merge inapoi si din back ul telefonului

File: TechTitans/Views/Components/User/SearchButton.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace TechTitans.Views.Components.User
+{
+    public partial class SearchSongsButton : ContentView
+    {
+        public SearchSongsButton()
+        {
+            InitializeComponent();
+        }
+
+        private void OnSearchClick(object sender, EventArgs e) => Application.Current.MainPage = new NavigationPage(new SearchPage());
```
Comment (mihneadumi) [inline TechTitans/Views/Components/User/SearchButton.xaml.cs:16]: here it should be: so you can remove the back button completely from the Page
private void OnSearchClick(object sender, EventArgs e) => Navigation.PushAsync(new SearchPage());

File: TechTitans/Views/Components/User/BackToUser.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+namespace TechTitans.Views.Components.User
+{
+    public partial class BackToUserButton : ContentView
+    {
+        public BackToUserButton()
+        {
+            InitializeComponent();
+        }
+
+        private void OnBackClick(object sender, EventArgs e) => Application.Current.MainPage = new NavigationPage(new UserPage());
```
Comment (mihneadumi) [inline TechTitans/Views/Components/User/BackToUser.xaml.cs:16]: nu i nevoie daca navighezi cu PushAsync, poti scoate si butonul din .xaml

File: TechTitans/Views/EndOfYearRecap.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                Artist = "BDLP",
+                Features = new[] {"Ian"},
+                Language = "Romanian",
+                Country = "Romania",
+                Album = "Single",
+                Image = "https://i.ytimg.com/vi/Ovbn5mPit8o/sddefault.jpg?v=64c3f573"
+            },
+            new()
+            {
+                SongId = 1,
```
Comment (mihneadumi) [inline TechTitans/Views/EndOfYearRecap.xaml.cs:54]: nu e neaparat sa fixezi dar aici e indentare ciudata prob de la copilot cand a generat idk

Comment (mihneadumi) [reply TechTitans/Views/EndOfYearRecap.xaml.cs:54]: asta e de la dani de aia zic ca nu i neaparat

File: TechTitans/Views/SearchPage.xaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                Placeholder="find a song..."
+                TextChanged="SearchBar_TextChanged">
+            </SearchBar>
+            <VerticalStackLayout Spacing="15" Padding="10, 0">
+                <Grid x:Name="SongsGrid">
+                    <Grid.RowDefinitions>
+                        <RowDefinition Height="*"/>
+                        <RowDefinition Height="*" />
+                        <RowDefinition Height="*" />
+                    </Grid.RowDefinitions>
```
Comment (mihneadumi) [inline TechTitans/Views/SearchPage.xaml:23]: daca iti definesti numar fix de rows in .xaml cand depasesti 6 song uri o sa inceapa sa le suprapuna pe ultimul rand.Trebuie adaugate dinamic cand adaugi si song urile, poti sterge asta, iti scriu o sugestie de fix in SearchPage.xaml.cs

File: TechTitans/Views/SearchPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        public SearchPage() {
+            InitializeComponent();
+        }
+        int rowIndex = 0;
+        int columnIndex = 0;
+        private void SearchBar_TextChanged(object sender, EventArgs e) { 
+        //here we should search from our database with some function (in repo/service) but will do 
+        //a mock function just for frontend development puposes
+            string song_to_find=((SearchBar)sender).Text;
+            var songs = new ObservableCollection<SongBasicInfo>(GetSongs(song_to_find));
```
Comment (mihneadumi) [inline TechTitans/Views/SearchPage.xaml.cs:25]: aici trb introdus un
SongsGrid.RowDefinitions.Add(new RowDefinition());
ca sa adaugi randul initial

File: TechTitans/Views/UserPage.xaml [hunk 6]
```diff
@@ -4,16 +4,72 @@
...
+            HorizontalOptions="Start" 
+            Margin="10,0,0,0"
+                />
+            <VerticalStackLayout Spacing="15" Padding="10, 0">
+                <Grid x:Name="SongsAdvertisedGrid">
+                    <Grid.RowDefinitions>
+                        <RowDefinition Height="*"/>
+                        <RowDefinition Height="*" />
+                        <RowDefinition Height="*" />
+                    </Grid.RowDefinitions>
```
Comment (mihneadumi) [inline TechTitans/Views/UserPage.xaml:65]: aceeasi situatie ca in search page

**Pull request merged**